src/main/java/de/muenchen/isi/domain/mapper/SearchDomainMapper.java

```diff
@@ -1,73 +1,192 @@
+            @Mapping(target = "coordinate", ignore = true),
+        }
+    )
+    public abstract AbfrageSearchResultModel entity2SearchResultModel(final WeiteresVerfahren entity);
+
+    @AfterMapping
+    public void afterMappingEntity2SearchResultModel(
+        final WeiteresVerfahren entity,
+        @MappingTarget final AbfrageSearchResultModel model
+    ) {
+        model.setCoordinate(this.getCoordinateFromAdresseOrVerortung(entity.getAdresse(), entity.getVerortung()));
+    }
+
+    @Mappings(
+        {
+            @Mapping(target = "type", constant = SearchResultType.Values.INFRASTRUKTUREINRICHTUNG),
+            @Mapping(target = "coordinate", ignore = true),
         }
     )
-    AbfrageSearchResultModel entity2SearchResultModel(final WeiteresVerfahren entity);
+    public abstract InfrastruktureinrichtungSearchResultModel entity2SearchResultModel(
+        final Infrastruktureinrichtung entity
+    );
 
-    @Mappings({ @Mapping(target = "type", constant = SearchResultType.Values.INFRASTRUKTUREINRICHTUNG) })
-    InfrastruktureinrichtungSearchResultModel entity2SearchResultModel(final Infrastruktureinrichtung entity);
+    @AfterMapping
+    public void afterMappingEntity2SearchResultModel(
+        final Infrastruktureinrichtung entity,
+        @MappingTarget final InfrastruktureinrichtungSearchResultModel model
+    ) {
+        if (hasAdressCoordinate(entity.getAdresse())) {
+            model.setCoordinate(koordinatenDomainMapper.entity2Model(entity.getAdresse().getCoordinate()));
+        } else if (ObjectUtils.isNotEmpty(entity.getVerortung())) {
+            if (ObjectUtils.isNotEmpty(entity.getVerortung().getPoint())) {
+                WGS84Model wgs84Model = new WGS84Model();
+                wgs84Model.setLongitude(entity.getVerortung().getPoint().getCoordinates().get(0).doubleValue());
+                wgs84Model.setLatitude(entity.getVerortung().getPoint().getCoordinates().get(1).doubleValue());
+                model.setCoordinate(wgs84Model);
+            }
+        } else {
+            model.setCoordinate(null);
+        }
+    }
 
-    default UUID map(final Bauvorhaben bauvorhaben) {
+    public UUID map(final Bauvorhaben bauvorhaben) {
         return ObjectUtils.isEmpty(bauvorhaben) ? null : bauvorhaben.getId();
     }
+
+    public boolean hasAdressCoordinate(final Adresse adresse) {
+        if (ObjectUtils.isNotEmpty(adresse)) {
+            if (ObjectUtils.isNotEmpty(adresse.getCoordinate())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public boolean hasVerortungCoordinate(final VerortungMultiPolygon verortung) {
+        if (ObjectUtils.isNotEmpty(verortung)) {
+            if (ObjectUtils.isNotEmpty(verortung.getMultiPolygon())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public WGS84Model getCoordinateFromAdresseOrVerortung(
```

Comment:
In Javadoc für die Methode  `getCoordinateFromAdresseOrVerortung` dokumentieren, wann aus aus welchem Parameter die Puinktkoordinate ermittelt wird. 

Alternativ den Methodennamen sprechender machen. Evtl: `getCoordinateFromAdresseOrFromVerortungWhenNoAdressIsGiven`